### DISCUSSION TITLE
application crash

### DISCUSSION THREAD

**Discussion by tobias-carlbom** [jchristn/WatsonTcp#258] [Q&A]
Hi,

My application sometimes crashes and i get the error below in event viewer:

Application: wqruntime.exe
Framework Version: v4.0.30319
Description: The process was terminated due to an unhandled exception.
Exception Info: exception code c0000005, exception address 00007FFFED6DD131

I am using watsontcp 4.8.14.14 in a class library. The class library is then used in another program which has a c# api, e.g. the program can interact with c# class library. Newer versions of watsontcp does not seem compatible with the program which as the c# api.

The crash seem to happen frequently when a client connects directly after the watsontcp server starts.

I have activated DebugMessages and Logger, but there are no relevent information where.

Do you guys have any idea of how to troubleshoot this?

**Comment by tobias-carlbom** [comment]
I got verified that the .net api of the external program was recently updated to .net framework 4.8.

So i will need to update the class library .net framework from 4.7.2 to 4.8. I guess that this means that i also need to update all dependencies of the project :/

**Reply by jchristn** [reply]
Hi, are you certain that the crash is being caused by this library?  Could you also set handlers for both server and client ```Events.ExceptionEncountered```?

Also, have you tried using the latest version?

Any more details on the incompatibility you mention would be helpful.  (You had mentioned: "I am using watsontcp 4.8.14.14 in a class library. The class library is then used in another program which has a c# api, e.g. the program can interact with c# class library. Newer versions of watsontcp does not seem compatible with the program which as the c# api.")

Cheers, Joel
